fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -240,7 +241,14 @@ private void registerMaterializedViews(OptExpression logicOperatorTree, ConnectC
             // generate scan mv plan
             Database db = context.getCatalog().getDb(mv.getDbId());
             TableName tableName = new TableName(db.getFullName(), mv.getName());
-            String selectMvSql = "select * from " + tableName.toSql();
+            StringBuilder selectMvSb = new StringBuilder("select * from " + tableName.toSql());
+
+            ScalarOperator predicate = mvOptimizer.getPartialPartitionPredicate(mv);
+            if (predicate != null) {
+                selectMvSb.append(" where ");
```

Comment:
because when some partitions are out dated. the scan should only read part of the mv data. here add where to limit the part. and which will do partition prune.